mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+        of the detection bboxes. 2. Whether ignore the classes that unannotated
+        on that image.
+        """
+        if image_level_annotations is not None:
+            assert len(annotations) == \
+                   len(image_level_annotations) == \
+                   len(det_results)
+        else:
+            assert len(annotations) == len(det_results)
+        for i in range(len(det_results)):
+            results = copy.deepcopy(det_results[i])
+            valid_classes = np.where(
+                np.array([[bbox.shape[0]] for bbox in det_results[i]]) != 0)[0]
+            if image_level_annotations is not None:
+                labels = annotations[i]['labels']
+                image_level_labels = \
+                    image_level_annotations[i]['image_level_labels']
+                allowed_labeles = np.unique(
+                    np.append(labels, image_level_labels))
+            else:
+                allowed_labeles = np.unique(annotations[i]['labels'])
+
+            for valid_class in valid_classes:
+                det_cls = np.where(self.class_label_tree[valid_class])[0]
+                for index in det_cls:
+                    if index in allowed_labeles and \
+                            index != valid_class and \
+                            self.get_supercategory:
+                        det_results[i][index] = \
+                            np.concatenate((det_results[i][index],
+                                            results[valid_class]))
+                    elif index not in allowed_labeles and self.filter_labels:
+                        # Remove useless parts
+                        det_results[i][index] = np.empty(
+                            (0, 5)).astype(np.float32)
+        return det_results
+
+    def load_image_label_from_csv(self, image_level_ann_file):
+        """Load image level annotations from csv style ann_file.
+
+        Args:
+            image_level_ann_file (str): CSV style image level annotation
+                file path.
+
+
+        Returns:
+            defaultdict[list[dict]]: Annotations where item of the defaultdict
+            indicates an image, each of which has (n) dicts.
+            Keys of dicts are:
+
+                - `image_level_label` (int): Label id.
+                - `confidence` (float): Labels that are human-verified to be
+                    present in an image have confidence = 1 (positive labels).
+                    Labels that are human-verified to be absent from an image
+                    have confidence = 0 (negative labels). Machine-generated
+                    labels have fractional confidences, generally >= 0.5.
+                    The higher the confidence, the smaller the chance for
+                    the label to be a false positive.
+        """
+
+        item_lists = defaultdict(list)
+        with open(image_level_ann_file, 'r') as f:
+            reader = csv.reader(f)
+            i = -1
```

Comment:
the use of i is a little bit ugly. Considering use a flag like first_line or enumerate reader and skip i == 0